### PR TITLE
Coded for TC01

### PR DESCRIPTION
**Pull request by lqthai07t3**
Pull Request for TC01. Please review for me. Thank you so much.

### CODE REVIEW

File: src/main/java/pages/BasePage.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package pages;
```
Comment (tanmle) [inline src/main/java/pages/BasePage.java:1]: do you need everything in this BasePage for just a login test case? Remember! Just push what you need for the test cases.

File: src/main/java/pages/HomePage.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import helpers.DriverHelper;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class HomePage extends BasePage{
+
+    //locator
+    private By lblWelcomeUser = By.cssSelector("div.account>strong");
+
```
Comment (tanmle) [inline src/main/java/pages/HomePage.java:11]: There are 2 downlines here, 1 is enough.

File: src/main/java/pages/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    //WebElement
+    private WebElement getLblWelcomeUser(){ return DriverHelper.getDriver().findElement(lblWelcomeUser); }
+
+
+    //Methods
+    public String getWelcomeUserMessage(){
+        return getLblWelcomeUser().getText().trim();
+    }
+
```
Comment (tanmle) [inline src/main/java/pages/HomePage.java:21]: Unneeded downline.

File: src/main/java/pages/LoginPage.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    private WebElement getTxtPassword(){ return DriverHelper.getDriver().findElement(txtPassword); }
+    private WebElement getBtnLogin(){ return DriverHelper.getDriver().findElement(btnLogin); }
+
+    //Methods
+    public void login(String email,String password){
+        getTxtEmail().sendKeys(email);
+        getTxtPassword().sendKeys(password);
+        getBtnLogin().click();
+    }
+
```
Comment (tanmle) [inline src/main/java/pages/LoginPage.java:25]: Unneeded downlines

File: src/test/java/tests/LoginTests.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        loginPage.login(Constants.USERNAME, Constants.PASSWORD);
+
+        //DriverHelper.getDriver().manage().window().setSize(new Dimension(1080,780));
+
+        LogHelper.info("Verify:");
+        Assert.assertTrue((homePage.getWelcomeUserMessage().equals("Welcome "+Constants.USERNAME)) && (loginPage.isLogoutTabDisplayed()),
+                "User can not log in successfully with valid account"
+        );
+    }
+
```
Comment (tanmle) [inline src/test/java/tests/LoginTests.java:32]: Unneeded downlines.

File: src/test/java/tests/LoginTests.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    @Test(testName = "User can log into Railway with valid username and password")
+    public void tc001_LoginWithValidUsernameAndPassword() {
+        LogHelper.info("Go to Login Tab");
+        loginPage.clickLoginTab();
+
+        LogHelper.info("Log in with already register account");
+        loginPage.login(Constants.USERNAME, Constants.PASSWORD);
+
+        //DriverHelper.getDriver().manage().window().setSize(new Dimension(1080,780));
```
Comment (tanmle) [inline src/test/java/tests/LoginTests.java:25]: why don't you remove this line?

File: src/test/java/tests/LoginTests.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+        LogHelper.info("Verify:");
```
Comment (tanmle) [inline src/test/java/tests/LoginTests.java:27]: What `Verify`?

File: src/test/java/tests/LoginTests.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        Assert.assertTrue((homePage.getWelcomeUserMessage().equals("Welcome "+Constants.USERNAME)) && (loginPage.isLogoutTabDisplayed()),
```
Comment (tanmle) [inline src/test/java/tests/LoginTests.java:28]: Code format

File: src/main/java/pages/BasePage.java (hunk 1 above)
Comment (lqthai07t3) [reply src/main/java/pages/BasePage.java:1]: Okie, I fixed it

File: src/main/java/pages/LoginPage.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
+package pages;
+
+import helpers.DriverHelper;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class LoginPage extends BasePage{
+
+    //Locators
```
Comment (MinhQuan119) [inline src/main/java/pages/LoginPage.java:9]: Please use the same below format for all pages, I saw that they are not consistence
```
//Locators

//WebElements

//Methods
```

**Review by MinhQuan119 — CHANGES_REQUESTED** [review]

File: src/main/java/pages/HomePage.java (hunk 3 above)
Comment (lqthai07t3) [reply src/main/java/pages/HomePage.java]: Removed unneeded downline code

File: src/main/java/pages/LoginPage.java (hunk 4 above)
Comment (lqthai07t3) [reply src/main/java/pages/LoginPage.java]: Removed unneeded downline code

File: src/test/java/tests/LoginTests.java (hunk 5 above)
Comment (lqthai07t3) [reply src/test/java/tests/LoginTests.java]: Removed unneeded downline code

File: src/test/java/tests/LoginTests.java (hunk 6 above)
Comment (lqthai07t3) [reply src/test/java/tests/LoginTests.java]: Removed redundant code

File: src/test/java/tests/LoginTests.java (hunk 7 above)
Comment (lqthai07t3) [reply src/test/java/tests/LoginTests.java]: Add information to verify.

File: src/test/java/tests/LoginTests.java (hunk 8 above)
Comment (lqthai07t3) [reply src/test/java/tests/LoginTests.java]: Formated code

File: src/test/java/tests/LoginTests.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    @Test(testName = "User can log into Railway with valid username and password")
+    public void tc001_LoginWithValidUsernameAndPassword() {
+        LogHelper.info("Go to Login Tab");
+        loginPage.clickLoginTab();
+
+        LogHelper.info("Log in with already register account");
+        loginPage.login(Constants.USERNAME, Constants.PASSWORD);
+
+        LogHelper.info("Verify: Welcome username message displays correctly and LogoutTab displays");
+        Assert.assertTrue((homePage.getWelcomeUserMessage().equals("Welcome " + Constants.USERNAME)) && (loginPage.isLogoutTabDisplayed()),
```
Comment (anhlvt) [inline src/test/java/tests/LoginTests.java:23]: Should separate to 2 Assertion

Comment (lqthai07t3) [reply src/test/java/tests/LoginTests.java]: Separated to 2 Assertion for TC01

File: src/main/java/pages/HomePage.java (hunk 2 above)
Comment (lqthai07t3) [reply src/main/java/pages/HomePage.java:10]: Fixed it, please check again

File: src/main/java/pages/LoginPage.java (hunk 9 above)
Comment (lqthai07t3) [reply src/main/java/pages/LoginPage.java:8]: I fixed it, but it's still here. Please check again

**Review by MinhQuan119 — APPROVED** [review]

**Review by tanmle — APPROVED** [review]

**Pull request merged**